### PR TITLE
feat(map): implement map.jinja from template-formula

### PR DESCRIPTION
**Pull request by sticky-note**
Implements map.jinja from template-formula

### CODE REVIEW

**Comment by myii** [comment]
Excellent work, @sticky-note.  I'll just run some tests to make sure that the rendered map is still the same, of course including the defaults.

File: openldap/defaults.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+---
+openldap:
+  client_config:   /etc/ldap.conf
+  client_pkg:      openldap
+  server_defaults: /etc/default/slapd
+  server_pkg:      openldap
+  service:         slapd
+  database:        mdb
+  su_group:        root
+  index:           objectClass
```
Comment (myii) [inline openldap/defaults.yaml]: While this layout is aesthetically more pleasing, it doesn't work with the `yamllint` defaults rules that we're using across formulas:

* https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.colons

Basically, we're using the default of _one space_ after colons.  That would apply to all of the files in this PR's changeset.

File: openldap/map.jinja [hunk 2]
```diff
@@ -1,97 +1,55 @@
...
+    )
+%}
+
+{%- set openldap = config %}
+
+{#- Post-processing for specific non-YAML customisations #}
+{%- if grains.os == 'MacOS' %}
+{%-   set macos_group = salt['cmd.run']("stat -f '%Sg' /dev/console") %}
+{%-   do TEMPLATE.update({'rootgroup': macos_group}) %}
+{%- endif %}
```
Comment (myii) [inline openldap/map.jinja]: If we're going to keep this section, then `TEMPLATE` needs to be updated accordingly.  However, there's no indication if this formula supports `MacOS` in the first place, so perhaps best to remove this block for the time being.

File: openldap/osfingermap.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# If you do not need to provide defaults via the `os_finger` grain,
+# you will need to provide at least an empty dict in this file, e.g.
+# osfingermap: {}
+---
+# os: Ubuntu
+Ubuntu-18.04: {}
+
+# os: CentOS
+CentOS-6: {}
+CentOS-7: {}
```
Comment (myii) [inline openldap/osfingermap.yaml]: Ouch, that's actually wrong in the `template-formula` itself.  It's actually `CentOS Linux-7` (`CentOS-6` is correct).

File: openldap/server.sls [hunk 4]
```diff
@@ -13,6 +13,8 @@
    - makedirs: True
    - require:
      - pkg: {{ openldap.server_pkg }}
+   - context:
+      openldap: {{ openldap | json }}
```
Comment (myii) [inline openldap/server.sls]: Missing one space here (should be a four-space indent):

```suggestion
       openldap: {{ openldap | json }}
```

File: openldap/osfamilymap.yaml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  server_pkg:       openldap2
+  service:          slapd
+  slapd_includes:
+    - /etc/openldap/schema/core.schema
+    - /etc/openldap/schema/cosine.schema
+    - /etc/openldap/schema/inetorgperson.schema
+    - /etc/openldap/schema/rfc2307bis.schema
+    - /etc/openldap/schema/yast.schema
+  slapd_argsfile:   /var/run/slapd/slapd.args
+  slapd_pidfile:    /var/run/slapd/slapd.pid
```
Comment (myii) [inline openldap/osfamilymap.yaml]: The lines for the original `map.jinja` are:

```jinja
        'slapd_argsfile': '/var/run/openldap/slapd.args',
        'slapd_pidfile': '/var/run/openldap/slapd.pid',
```

Do you have a reference for this fix?

Comment (myii) [reply openldap/osfamilymap.yaml]: Looks like https://www.suse.com/c/ldap-proxy-server-multiple-sources/ is a potential reference.

**Comment by myii** [comment]
Otherwise, this is working well (tested on a `Debian`-based system):

```diff
--- Before
+++ After
@@ -1,6 +1,8 @@
+arch: amd64
 client_config: /etc/ldap.conf
 client_pkg: ldap-utils
 database: mdb
+index: objectClass
 server_config: /etc/ldap/slapd.conf
 server_defaults: /etc/default/slapd
 server_pkg: slapd
```

File: openldap/osfingermap.yaml (hunk 3 above)
Comment (myii) [reply openldap/osfingermap.yaml]: I've added a fix for this in https://github.com/saltstack-formulas/template-formula/pull/184.

File: openldap/defaults.yaml (hunk 1 above)
Comment (sticky-note) [reply openldap/defaults.yaml]: done

File: openldap/server.sls (hunk 4 above)
Comment (sticky-note) [reply openldap/server.sls]: done

File: openldap/osfamilymap.yaml (hunk 5 above)
Comment (sticky-note) [reply openldap/osfamilymap.yaml]: here is another reference -> https://www.pks.mpg.de/~mueller/docs/suse10.1/suselinux-manual_en/manual/sec.ldap.slapd.html
There was duplicate 'slapd_pidfile' key in the old `map.jinja`, the second entry was `'slapd_pidfile': '/var/run/slapd/slapd.pid'`. I think someone wanted to correct the path but added a duplicate key.

File: openldap/map.jinja (hunk 2 above)
Comment (sticky-note) [reply openldap/map.jinja]: done

**Comment by sticky-note** [comment]
@myii Force pushed changes. It´s okay for me if it is ok for you

**Comment by myii** [comment]
Merged, thanks for going to all of the effort, @sticky-note.

**Pull request merged**